### PR TITLE
feat(components): add ecc-ui-utils package

### LINKED ISSUE #50: feat: Package for reusable components that can be abstracted for other components
**Please describe your issue.**

This package provides a collection of reusable components that can be easily incorporated into other components. By utilizing these components, it guarantees enhanced reusability, scalability, and consistency in component design. These abstracted components enable developers to efficiently build and maintain applications with consistent design patterns and promote modular development practices.

**Acknowledgement**

> We thank you for your valuable contribution and would like to acknowledge it
> on our [contributors][contributors] page. To help us with that, we would like
> to kindly request that you help us with this by commenting on this issue with
> a request to the [@all-contributors bot][all-contributors-bot] of the form
> `@all-contributors please add @YOUR_GH_HANDLE for TYPE_1, TYPE_2, ...`, where
> `TYPE_1` etc. refer to [contribution types supported by the All Contributors
> Specification][all-contributors-types].

[all-contributors-bot]: <https://allcontributors.org/docs/en/bot/overview>
[all-contributors-types]: <https://allcontributors.org/docs/en/emoji-key>
[contributors]: ../../CONTRIBUTORS

### PR DESCRIPTION
**Pull request by JaeAeich**
## Description

This PR is the initial setup/config for the above mentioned package using official FAST CLI. 
-

Fixes #50

## Checklist

> Please go through the following checklist to ensure that your change is ready
> for review. Please do not forget to double check the list after you have
> modified your PR, e.g., if you have added commits to address reviewer
> comments or to fix failing automated checks. Please check items also if they
> do not apply to your change, e.g., if your change does not require an update
> of the user-facing documentation, then still check the box. Generally, PRs
> are only reviewed when all boxes are ticked off and all automated checks pass
> (use the comment section below if you believe that your PR is ready to be
> merged even though not all boxes were ticked off).

- [x] My code follows the [contributing guidelines][contributing] of this
      project, including, in particular, with regard to any style guidelines
- [x] The title of my PR complies with the [Conventional Commits
      specification][conventional-commits]; in particular, it clearly indicates
      that a change is a breaking change
- [x] I am aware that all my commits will be squashed into a single commit,
      using the PR title as the commit message.
- [x] I have performed a self-review of my own code
- [x] I have commented my code in hard-to-understand areas
- [ ] I have updated the user-facing documentation to describe any new or
      changed behavior
- [ ] I have added type annotations for all function/class/method interfaces
      or updated existing ones
- [ ] I have provided [JSDoc][jsdoc] block tags for all
      functions/classes/methods or updated existing ones
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature
      works
- [x] New and existing unit tests pass locally with my changes
- [x] I have not reduced the existing code coverage
- [ ] I have asked the [@all-contributors bot][all-contributors-bot] to
      acknowledge my contributions by commenting on this PR with a request of
      the form `@all-contributors please add @YOUR_GH_HANDLE for TYPE_1,
    TYPE_2, ...`, where `TYPE_1` etc. refer to [contribution types supported
      by the All Contributors Specification][all-contributors-types] OR I do
      not want my contributions to be acknowledged

## Comments
unchecked box not required

[all-contributors-bot]: https://allcontributors.org/docs/en/bot/overview
[all-contributors-types]: https://allcontributors.org/docs/en/emoji-key
[contributing]: CONTRIBUTING.md
[conventional-commits]: https://www.conventionalcommits.org/en/v1.0.0/
[jsdoc]: https://jsdoc.app/index.html

### CODE REVIEW

**Comment by JaeAeich** [comment]
@git-anurag-hub @SalihuDickson Please check/review the config; I have integrated eslint package and changed some ts configs of the boiler code FAST CLI provided. If this is right, then I would do the same for the `tes` package.

**Comment by git-anurag-hub** [comment]
Hey @JaeAeich, please merge the `main` branch into this branch & resolve conflicts.

File: packages/ecc-ui-utils/.gitignore
Comment (SalihuDickson) [inline packages/ecc-ui-utils/.gitignore:1]: your gitignore looks a little thin, I think you can just go ahead and copy the gitgnore from your other PR, the ecc-client-ga4gh-tes package.

File: packages/ecc-ui-utils/tsconfig.json
Comment (SalihuDickson) [inline packages/ecc-ui-utils/tsconfig.json:1]: there are some config options you should add here for development purposes, like the sourceMap and InlineSources. Check and make sure this file has all the config options it should.

**Review by SalihuDickson — CHANGES_REQUESTED** [review]
You need to merge the changes from the main branch to resolve the conflicts in the cookiecutter,config.js file.

**Comment by git-anurag-hub** [comment]
Also @JaeAeich, fix the naming of the package. It doesn't follow the proper guidelines!

**Review by git-anurag-hub — CHANGES_REQUESTED** [review]
@JaeAeich @ltwmori, please merge the main branch to it & resolve. 
Also, @JaeAeich fix the naming of the package. It doesn't follow the proper guidelines! 
Please solve them ASAP! 🙏🏼

**Comment by git-anurag-hub** [comment]
Also @uniqueg @JaeAeich @SalihuDickson @ltwmori, let's reconsider the need to create a separate package for this use case. It appears that the functionality provided by this package aligns with the existing `elixir-ui` package (which we can rename to a more generic one). Some points to be noted:
- Instead of maintaining two separate packages, we can handle the UI component within `elixir-ui` and make it available for use by other packages as well. 
- This approach offers the advantage of having a single configuration for all components, reducing unnecessary duplication and complexity in work `package-author` work. 
- Additionally, it eliminates the need to write redundant logic in two separate places. By consolidating our efforts into one package, we can streamline development and improve overall efficiency.

**Comment by SalihuDickson** [comment]
> Also @uniqueg @JaeAeich @SalihuDickson @ltwmori, let's reconsider the need to create a separate package for this use case. It appears that the functionality provided by this package aligns with the existing `elixir-ui` package (which we can rename to a more generic one). Some points to be noted:
> 
> * Instead of maintaining two separate packages, we can handle the UI component within `elixir-ui` and make it available for use by other packages as well.
> * This approach offers the advantage of having a single configuration for all components, reducing unnecessary duplication and complexity in work `package-author` work.
> * Additionally, it eliminates the need to write redundant logic in two separate places. By consolidating our efforts into one package, we can streamline development and improve overall efficiency.

Yeah, these are good points. One thing though, the elixir-ui package is configured with FAST, will that not cause integration issues with other frameworks?

**Comment by JaeAeich** [comment]
Yeah @SalihuDickson but if that is the case (which I don't think it is), even `ecc-ui-utils` is configured with FAST, I think @git-anurag-hub 's points are valid this package would just play as an unnecessary overhead when it can be integrated into `elixir-ui` as they both are pushing to offer the same feature.

**Comment by SalihuDickson** [comment]
> Yeah @SalihuDickson but if that is the case (which I don't think it is), even `ecc-ui-utils` is configured with FAST, I think @git-anurag-hub 's points are valid this package would just play as an unnecessary overhead when it can be integrated into `elixir-ui` as they both are pushing to offer the same feature.

Yh, i wasn't disagreeing with him on that.

**Comment by JaeAeich** [comment]
@uniqueg @git-anurag-hub @SalihuDickson Should I close this PR? (please refer to the above discussion)

**Comment by uniqueg** [comment]
> @uniqueg @git-anurag-hub @SalihuDickson Should I close this PR? (please refer to the above discussion)

I think so...

**Pull request closed without merging**